menus/settings/custom-fields-translation.php

```diff
@@ -0,0 +1,112 @@
+		<p><?php echo sprintf( __( 'The following table shows the custom fields with no translation preferences and suggests a translation preference for each field. You can review them and generate the XML. Once the XML is generated, you can copy it and paste it in the <a href="%s">Custom XML Configuration</a> tab in WPML.', 'wpml-compatibility-test-tools' ), esc_url( admin_url( 'admin.php?page=tm%2Fmenu%2Fsettings&sm=custom-xml-config' ) ) ); ?></p>
+	</div>
+
+	<form id="wpml-cf-form">
+		<table class="widefat">
+			<?php if ( $custom_fields ): ?>
+			<thead>
+			<tr>
+				<th><?php esc_html_e( 'Custom fields', 'wpml-compatibility-test-tools' ); ?></th>
+				<th><?php esc_html_e( "Don't translate", 'wpml-compatibility-test-tools' ); ?></th>
+				<th><?php esc_html_e( "Copy", 'wpml-compatibility-test-tools' ); ?></th>
+				<th><?php esc_html_e( "Copy once", 'wpml-compatibility-test-tools' ); ?></th>
+				<th><?php esc_html_e( "Translate", 'wpml-compatibility-test-tools' ); ?></th>
+				<th><?php esc_html_e( "Sample", 'wpml-compatibility-test-tools' ); ?></th>
+			</tr>
+			</thead>
+			<tbody class="wctt">
+			<?php foreach ( $translation_preferences as $custom_field => $translation_preference ): ?>
+				<tr>
+					<td>
+						<label>
+							<?php echo esc_html( $custom_field ); ?>
+						</label>
+					</td>
+					<td title="<?php esc_attr_e( "Don't translate", 'wpml-compatibility-test-tools' ); ?>">
+						<input id="cf_0_<?php echo esc_attr( $custom_field ); ?>" type="radio"
+						       name="cf[<?php echo esc_attr( $custom_field ); ?>]" value="ignore"/>
+					</td>
+					<td title="<?php esc_attr_e( "Copy", 'wpml-compatibility-test-tools' ); ?>">
+						<input id="cf_1_<?php echo esc_attr( $custom_field ); ?>" type="radio"
+						       name="cf[<?php echo esc_attr( $custom_field ); ?>]"
+						       value="copy" <?php checked( $translation_preference, 'copy' ); ?>/>
+					</td>
+					<td title="<?php esc_attr_e( "Copy once", 'wpml-compatibility-test-tools' ); ?>">
+						<input id="cf_2_<?php echo esc_attr( $custom_field ); ?>" type="radio"
+						       name="cf[<?php echo esc_attr( $custom_field ); ?>]" value="copy-once"/>
+					</td>
+					<td title="<?php esc_attr_e( "Translate", 'wpml-compatibility-test-tools' ); ?>">
+						<input id="cf_3_<?php echo esc_attr( $custom_field ); ?>" type="radio"
+						       name="cf[<?php echo esc_attr( $custom_field ); ?>]"
+						       value="translate" <?php checked( $translation_preference, 'translate' ); ?>/>
+					</td>
+					<td>
+						<?php
+						global $wpdb;
+
+						$value = $wpdb->get_var( $wpdb->prepare( "SELECT meta_value FROM $wpdb->postmeta WHERE meta_key = %s LIMIT 1", $custom_field ) );
+
+						// Define the maximum number of characters to display
+						$maxLength = 50;
+
+						// If the value length is greater than the defined maximum length,
+						// cut it down and append '...' to indicate that it's truncated
+						if ( strlen( $value ) > $maxLength ) {
```

Comment:
@amElnagdy This also throws an error when its null:
```suggestion
						if ( $value && strlen( $value ) > $maxLength ) {
```